realm/realm-library/src/main/cpp/CMakeLists.txt

```diff
@@ -120,7 +120,7 @@ set(WARNING_CXX_FLAGS "-Wall -Wextra -pedantic -Wno-long-long -Wno-variadic-macr
 -Wno-missing-field-initializers -Wmissing-declarations -Wno-error=uninitialized -Wno-error=maybe-uninitialized")
 set(REALM_COMMON_CXX_FLAGS "-DREALM_ANDROID -DREALM_HAVE_CONFIG -DPIC -pthread -fvisibility=hidden -std=c++14 -fsigned-char")
 if (build_SYNC)
-    set(REALM_COMMON_CXX_FLAGS "${REALM_COMMON_CXX_FLAGS} -DREALM_SYNC")
+    set(REALM_COMMON_CXX_FLAGS "${REALM_COMMON_CXX_FLAGS} -DREALM_ENABLE_SYNC=1")
```

Comment:
Then you can do `#if REALM_ENABLE_SYNC` instead of `#ifdef REALM_ENABLE_SYNC`. Using `#ifdef` is a bit dangerous as `-DREALM_ENABLE_SYNC=NO` still defines `REALM_ENABLE_SYNC`.